lib/srv/db/access_test.go

```diff
@@ -255,114 +247,134 @@ func (m *testModules) Features() modules.Features {
 	go c.proxyServer.Serve(c.mux.DB())
-	// Start handling Postgres connection on the database server.
+	// Start MySQL proxy server.
+	go c.proxyServer.ServeMySQL(c.mysqlListener)
+	// Start handling database client connections on the database server.
 	for conn := range c.proxyConn {
 		c.server.HandleConnection(conn)
 	}
 }
 
 // postgresClient connects to test Postgres through database access as a
 // specified Teleport user and database account.
-func (c *testContext) postgresClient(ctx context.Context, teleportUser, dbUser, dbName string) (*pgconn.PgConn, error) {
+func (c *testContext) postgresClient(ctx context.Context, teleportUser, dbService, dbUser, dbName string) (*pgconn.PgConn, error) {
 	return postgres.MakeTestClient(ctx, common.TestClientConfig{
 		AuthClient: c.authClient,
 		AuthServer: c.authServer,
 		Address:    c.mux.DB().Addr().String(),
 		Cluster:    c.clusterName,
 		Username:   teleportUser,
 		RouteToDatabase: tlsca.RouteToDatabase{
-			ServiceName: c.postgresDBServer.GetName(),
+			ServiceName: dbService,
 			Protocol:    defaults.ProtocolPostgres,
 			Username:    dbUser,
 			Database:    dbName,
 		},
 	})
 }
 
-func (c *testContext) startHandlingMySQLConnections() {
-	// Start test MySQL server.
-	go c.mysqlServer.Serve()
-	// Start MySQL proxy server.
-	go c.proxyServer.ServeMySQL(c.mysqlListener)
-	// Start handling MySQL connections on the database server.
-	for conn := range c.proxyConn {
-		c.server.HandleConnection(conn)
-	}
-}
-
 // mysqlClient connects to test MySQL through database access as a specified
 // Teleport user and database account.
-func (c *testContext) mysqlClient(teleportUser, dbUser string) (*client.Conn, error) {
+func (c *testContext) mysqlClient(teleportUser, dbService, dbUser string) (*client.Conn, error) {
 	return mysql.MakeTestClient(common.TestClientConfig{
 		AuthClient: c.authClient,
 		AuthServer: c.authServer,
 		Address:    c.mysqlListener.Addr().String(),
 		Cluster:    c.clusterName,
 		Username:   teleportUser,
 		RouteToDatabase: tlsca.RouteToDatabase{
-			ServiceName: c.mysqlDBServer.GetName(),
+			ServiceName: dbService,
 			Protocol:    defaults.ProtocolMySQL,
 			Username:    dbUser,
 		},
 	})
 }
 
+// createUserAndRole creates Teleport user and role with specified names
+// and allowed database users/names properties.
+func (c *testContext) createUserAndRole(ctx context.Context, userName, roleName string, dbUsers, dbNames []string) (types.User, types.Role, error) {
+	user, role, err := auth.CreateUserAndRole(c.tlsServer.Auth(), userName, []string{roleName})
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+	role.SetDatabaseUsers(types.Allow, dbUsers)
+	role.SetDatabaseNames(types.Allow, dbNames)
+	err = c.tlsServer.Auth().UpsertRole(ctx, role)
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+	return user, role, nil
+}
+
 // Close closes all resources associated with the test context.
 func (c *testContext) Close() error {
```

Comment:
nit: this never returns an error
if you remove the return type, you can use `t.Cleanup(testCtx.Close)` instead of a closure